lib/fog/aws/storage.rb

```diff
@@ -298,7 +298,7 @@ def request_params(params)
                 host = params.fetch(:cname, bucket_name)
               elsif path_style
                 path = bucket_to_path bucket_name, path
-              else
+              elsif !host.include?(bucket_name)
```

Comment:
I think this might have a few edge cases, if the bucketname happens to be part of the normal amazon aws url. For example, a bucket named `north` would match `s3.ap-northeast-1.amazonaws.com`, even if the host didn't have the bucket name in it. Maybe we should specifically check to see if the host starts with the bucket_name to avoid this? (there are only a very few cases where what you did wouldn't work, but better safe than sorry). Does that make sense?

Reply:
Makes sense, let me make that change 👍